hnn_core/network.py

```diff
@@ -135,34 +136,62 @@ class Network(object):
     Parameters
     ----------
     params : dict
-        The parameters
+        The parameters to use for constructing the network.
 
     Attributes
     ----------
     params : dict
-        The parameters
+        The parameters of the network
     cellname_list : list
         The names of real cell types in the network (e.g. 'L2_basket')
     feedname_list : list
         The names of external drivers ('feeds') to the network (e.g. 'evdist1')
     gid_dict : dict
-        Dictionary with keys 'evprox1', 'evdist1' etc.
-        containing the range of Cell IDs of different cell
-        (or input) types.
+        A dictionary of unique identifiers of each real and artificial cell
+        in the network. Every cell type is represented by a key read from
+        cellname_list, followed by entries read from feedname_list. The value
+        of each key is a range of ints, one for each cell in given category.
+        Examples: 'L2_basal': range(0, 270), 'evdist1': range(272, 542), etc
+    pos_dict : dict
+        Dictionary containing the coordinate positions of all cells.
+        Keys are 'L2_pyramidal', 'L5_pyramidal', 'L2_basket', 'L5_basket',
+        'common', or any of the elements of the cellname or feedname lists.
     spikes : Spikes
         An instance of the Spikes object.
     """
 
     def __init__(self, params):
         # Save the parameters used to create the Network
         self.params = params
+        # Initialise a dictionary of cell ID's, which get assigned when the
+        # network is constructed ('built') in NetworkBuilder
+        # We want it to remain in each Network object, so that the user can
+        # interrogate a built and simulated net. In addition, Spikes are
+        # attached to a Network during simulation---Network is the natural
+        # place to keep this information
+        # XXX rename gid_dict in future
+        self.gid_dict = dict()
+
+        # When computing the network dynamics in parallel, the nodes of the
+        # network (real and artificial cells) potentially get distributed
+        # on different host machines/threads. NetworkBuilder._gid_assign
+        # assigns each node, identified by its unique GID, to one of the
+        # possible hosts/threads for computations. _gid_list here contains
+        # the GIDs of all the nodes assigned to the current host/thread.
+        self._gid_list = []
+
+        # XXX Note that GIDs are already defined here at the Network-level
+        # All that's left for NetworkBuilder is then to:
+        # - _PC.set_gid2node(gid, rank)
+        # - _PC.cell(gid, nrn_netcon) (or _PC.cell(feed_cell.gid, nrn_netcon))
+
+        # dict with keys 'common', 'evdist1', ...
+        # XXX needs better semantics, here to get first pass working
+        # list (len == n_trials) of dicts (keys: src_type) of event times
+        self._tmp_feed_event_times = []
```

Comment:
document it in docstring not in comments :) it's more easy to find there.

Reply:
Totally. Here I'm using a really _bad_ variable name (`_tmp...`) to remind me to get rid of it! The whole idea of this series of PRs is to get rid of 'feeds' and move to 'drives'. Semantics aside, the variable here turned out to be a list-of-dict-of-list-of-list---not sure that's the way to go :) Much of the work here is in order to get the tests to pass, _i.e._, match 'HNN Classic' models.

Reply:
I understand, but I'd still avoid bad variable names and call it `_feed_event_times` :) Who knows maybe you get busy after this PR. Also I will insist on the docstring ;-)